lib/class-sp-api.php

```diff
@@ -266,7 +265,19 @@ public function request( $url = '', $method = 'GET', $body = '', $request_params
 				'body'   => $body,
 			)
 		);
-		$result         = sp_remote_request( $url, $request_params );
+
+		$result = sp_remote_request( $url, $request_params );
+
+		/**
+		 * Fires after a request is made to the API.
+		 *
+		 * @param array|WP_Error $result          The result of the request.
+		 * @param string         $url             The URL to send the request to.
+		 * @param string         $method          The method for the request. Defaults to GET.
+		 * @param string         $body            The body of the request.
+		 * @param array          $request_params  Additional parameters.
+		 */
+		do_action( 'sp_request_response', $result, $url, $method, $body, $request_params );
```

Comment:
Is there harm in making this a filter to make it more extendible?

Reply:
No harm but a filter servers a different purpose than an action. Currently, the nypost uses this action to fire something rather than to change its output.

I guess we need both?!

Reply:
I guess we can ignore this for now then until the need arises

Reply:
I'll introduce it in a separate pr. :)